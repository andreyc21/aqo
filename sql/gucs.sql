CREATE EXTENSION aqo;
SET aqo.mode = 'learn';
SET aqo.show_details = true;

CREATE TABLE t(x int);
INSERT INTO t (x) (SELECT * FROM generate_series(1, 100) AS gs);
ANALYZE t;

EXPLAIN (ANALYZE, VERBOSE, COSTS OFF, TIMING OFF, SUMMARY OFF)
	SELECT x FROM t;
EXPLAIN (ANALYZE, VERBOSE, COSTS OFF, TIMING OFF, SUMMARY OFF)
	SELECT x FROM t;

DROP EXTENSION aqo;
