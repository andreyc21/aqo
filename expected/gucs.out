CREATE EXTENSION aqo;
SET aqo.mode = 'learn';
SET aqo.show_details = true;
CREATE TABLE t(x int);
INSERT INTO t (x) (SELECT * FROM generate_series(1, 100) AS gs);
ANALYZE t;
EXPLAIN (ANALYZE, VERBOSE, COSTS OFF, TIMING OFF, SUMMARY OFF)
	SELECT x FROM t;
                   QUERY PLAN                   
------------------------------------------------
 Seq Scan on public.t (actual rows=100 loops=1)
   AQO not used
   Output: x
 Using aqo: true
 AQO mode: LEARN
 JOINS: 0
(6 rows)

EXPLAIN (ANALYZE, VERBOSE, COSTS OFF, TIMING OFF, SUMMARY OFF)
	SELECT x FROM t;
                   QUERY PLAN                   
------------------------------------------------
 Seq Scan on public.t (actual rows=100 loops=1)
   AQO: rows=100, error=0%
   Output: x
 Using aqo: true
 AQO mode: LEARN
 JOINS: 0
(6 rows)

DROP EXTENSION aqo;
